gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -99,6 +104,21 @@ QueryRequest buildQueryRequest(
     // 3. Add GroupBy
     addGroupByExpressions(builder, request);
 
+    columnSelections.stream()
+        .filter(not(ExpressionReader::isSimpleAttributeSelection))
+        .forEach(
+            expression ->
+                requestContext.mapAliasToMapAttributeExpression(
+                    expression.getAttributeExpression().getAlias(),
+                    expression.getAttributeExpression()));
+    request.getGroupByList().stream()
```

Comment:
Do we care about group by if we're not selecting it back? Can't recall if that's auto-fetched or something

Reply:
The assumption is group bys are included as part of selections. Look at the comment here https://github.com/hypertrace/gateway-service/blob/b505531c68f211fa9ed3dbbeda75daa3124d4479/gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java#L175

```
// The expectation from the Gateway service client is that they
// do not add the
// group by expressions to the selection expressions in the request
```

Reply:
ah, strange. gotcha